packages/portfolio-contract/src/plan-solve.ts

```diff
@@ -426,36 +441,80 @@ export const rebalanceMinCostFlowSteps = (
     lastChain = chosen.srcChain;
   }
 
-  const steps: MovementDesc[] = prioritized.map(({ edge, flow }) => {
-    Number.isSafeInteger(flow) ||
-      Fail`flow ${flow} for edge ${edge} is not a safe integer`;
-    const amount = AmountMath.make(graph.brand, BigInt(flow));
-
-    let details = {};
-    switch (edge.feeMode) {
-      case 'gmpTransfer':
-        // TODO: Rather than hard-code, derive from Axelar `estimateGasFee`.
-        // https://docs.axelar.dev/dev/axelarjs-sdk/axelar-query-api#estimategasfee
-        details = { fee: AmountMath.make(graph.feeBrand, 30_000_000n) };
-        break;
-      case 'gmpCall':
-        // TODO: Rather than hard-code, derive from Axelar `estimateGasFee`.
-        // https://docs.axelar.dev/dev/axelarjs-sdk/axelar-query-api#estimategasfee
-        details = { fee: AmountMath.make(graph.feeBrand, 30_000_000n) };
-        break;
-      case 'toUSDN': {
-        // NOTE USDN transfer incurs a fee on output amount in basis points
-        const usdnOut =
-          (BigInt(flow) * (10000n - BigInt(edge.variableFee))) / 10000n;
-        details = { detail: { usdnOut } };
-        break;
+  const padding = 3n; // 300% padding on gas estimates
+  const steps: MovementDesc[] = await Promise.all(
+    prioritized.map(async ({ edge, flow }) => {
+      Number.isSafeInteger(flow) ||
+        Fail`flow ${flow} for edge ${edge} is not a safe integer`;
+      const amount = AmountMath.make(graph.brand, BigInt(flow));
+
+      await null;
+      let details = {};
+      switch (edge.feeMode) {
+        case 'makeEvmAccount':
+          details = {
+            // XXX: hardcoding until we can verify axelar API is accurate for this
+            detail: { evmGas: 200_000_000_000_000n },
+            fee: AmountMath.make(
+              graph.feeBrand,
+              gasEstimator
+                ? (await gasEstimator.getFactoryContractEstimate(
+                    chainOf(edge.dest) as AxelarChain,
+                  )) * padding
+                : 30_000_000n,
+            ),
+          };
+          break;
+        case 'withdraw':
+          details = {
+            fee: AmountMath.make(
+              graph.feeBrand,
+              gasEstimator
+                ? (await gasEstimator.getWalletEstimate(
+                    chainOf(edge.dest) as AxelarChain,
+                  )) * padding
+                : 30_000_000n,
+            ),
+          };
+          break;
+        case 'supply':
+          details = {
+            fee: AmountMath.make(
+              graph.feeBrand,
+              gasEstimator
+                ? (await gasEstimator.getWalletEstimate(
+                    chainOf(edge.dest) as AxelarChain,
+                  )) * padding
+                : 30_000_000n,
+            ),
+          };
+          break;
```

Comment:
As noted, these cases appear to be identical:
```suggestion
        case 'withinEvmViaAxelar': {
          const estimate = await gasEstimator.getFactoryContractEstimate(
            chainOf(edge.dest) as AxelarChain,
          );
          details = {
            fee: AmountMath.make(graph.feeBrand, padFeeEstimate(estimate)),
          };
          break;
```

Reply:
supply and withdraw are currently have the same estimate but will be different in the future, collapsing the switch cases but keeping them as distinct feeModes

Reply:
You said in another reply that fees for same-chain transactions are calculated based on a message from _Agoric_ to that chain, so why would the **calculations** differ for into vs. out of a pool on the other chain?

Reply:
the total fee is for both fee_for_agoric_to_evm + execution_fee_on_evm
the first part is the same but the second part varies because supply and withdraw executions will run different code on evm

Reply:
> the total fee is for both fee_for_agoric_to_evm + execution_fee_on_evm the first part is the same but the second part varies because supply and withdraw executions will run different code on evm

That reads to me like different inputs to the **same** calculation rather than _different_ calculations—perhaps the gasEstimator method should take src and dest inputs? Please add a TODO to revisit this (even if the final determination is that it should not change).